circuitpython_build_tools/scripts/build_bundles.py

```diff
@@ -37,7 +36,13 @@
 from circuitpython_build_tools import build
 from circuitpython_build_tools import target_versions
 
-import pkg_resources
+import importlib.resources as importlib_resources
+
+if sys.version_info < (3, 8):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
+
```

Comment:
yikes. maybe we should start by seeing if we can move relevant github actions workflows to newer python. For instance, the circuitpython bundle uses 3.10 presently: https://github.com/adafruit/Adafruit_CircuitPython_Bundle/blob/main/.github/workflows/build.yml#L24 and newly cookie-cut libs use 3.11 https://github.com/adafruit/workflows-circuitpython-libs/blob/main/build/action.yml#L11

Reply:
Since v3.7 is no longer supported, we could bump the minimum python version to 3.8 and the changes to requirements.txt and corresponding import here would get simplified.

Not sure what the motivation is behind bumping the minimum supported python version. Is it meant to keep non-std deps low? Were you hoping for less complex imports?

Reply:
yeah specifically I was reacting to needing to _add_ this version dependent code. I don't know why the workflows here are at 3.7 while we have 3.10 and 3.11 where it's actually being used.

Reply:
Even if you upgrade the version used in workflows, the pkg supported version is still set to
https://github.com/adafruit/circuitpython-build-tools/blob/b49cd8413aeed17b61bc05fe2344523666ffc061/setup.py#L16

Again, python 3.7 has reached end-of-life in 2023-06-27 (see [PEP537](https://peps.python.org/pep-0537/)).